pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -287,11 +287,14 @@ const (
 
 // VolumeOptions specifies capacity and tags for a volume.
 type VolumeOptions struct {
-	CapacityGB       int
-	Tags             map[string]string
-	PVCName          string
-	VolumeType       string
-	AvailabilityZone string
+	CapacityGB        int
+	Tags              map[string]string
+	PVCName           string
+	VolumeType        string
+	ZonePresent       bool
```

Comment:
IMHO, I can approach it using one of the below ways:

A, riskier: infer from the empty string `AvailabilityZone` or `AvailabilityZones` that the optional parameter `zone` or `zones` wasn't present. As the Persistent Volume Claim parameters are provided as `map[string]string` the values `"zone": ""` or `"zones": ""` are perfectly valid values in Go lang even though K8s yaml file parsing shall never provide such values. So in case I relay on the K8s yaml file parsing I can use the `AvailabilityZone` or `AvailabilityZones` empty string values as an indication that the `zone` or `zones` parameter were not specified in a Persistent Volume Claim. However, if my code is "fed" with unexpected input the incorrect input value `"zone": ""` or `"zones": ""` will remain undetected.

B, defensive: use a dedicated booleans to indicate whether the `zone` or `zones` optional parameters were present or not.

I chose the defensive approach.